image_data_guide/01_download_data.ipynb

```diff
@@ -0,0 +1,546 @@
+   "metadata": {},
+   "source": [
+    "## Sample the Dataset\n",
+    "___\n",
+    "In order to make working with the data easier, we'll select 250 images from each class at random. To make sure you get the same set of cell images for each run of this we'll also set Numpy's random seed to 0. This is a small fraction of the dataset, but it demonstrates how using transfer learning can give you good results without needing very large datasets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.random.seed(0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_annos = {}\n",
+    "\n",
+    "for category_id in category_labels:\n",
+    "    subset = [k for k, v in all_annos.items() if v['category_id'] == category_id]\n",
+    "    sample = np.random.choice(subset, size=250, replace=False)\n",
+    "    for k in sample:\n",
+    "        sample_annos[k] = all_annos[k]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create a download function\n",
+    "In order to parallelize downloading the images we must wrap the download and save process with a function for multi-threading with joblib."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def download_image(url, path):\n",
+    "    data = imread(url)\n",
+    "    imwrite(path / url.split('/')[-1], data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Download the sample of the dataset (2,500 images, ~5min)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_dir = pathlib.Path('data_sample_2500')\n",
+    "sample_dir.mkdir(exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[Parallel(n_jobs=5)]: Using backend ThreadingBackend with 5 concurrent workers.\n",
+      "[Parallel(n_jobs=5)]: Done  22 tasks      | elapsed:    2.2s\n",
+      "[Parallel(n_jobs=5)]: Done 118 tasks      | elapsed:   10.8s\n",
+      "[Parallel(n_jobs=5)]: Done 278 tasks      | elapsed:   25.1s\n",
+      "[Parallel(n_jobs=5)]: Done 502 tasks      | elapsed:   44.0s\n",
+      "[Parallel(n_jobs=5)]: Done 790 tasks      | elapsed:  1.2min\n",
+      "[Parallel(n_jobs=5)]: Done 1142 tasks      | elapsed:  1.7min\n",
+      "[Parallel(n_jobs=5)]: Done 1558 tasks      | elapsed:  2.4min\n",
+      "[Parallel(n_jobs=5)]: Done 2038 tasks      | elapsed:  3.2min\n",
+      "[Parallel(n_jobs=5)]: Done 2500 out of 2500 | elapsed:  3.9min finished\n"
+     ]
+    }
+   ],
+   "source": [
+    "with parallel_backend('threading', n_jobs=5):\n",
```

Comment:
This error showed up when running this cell:  `timeout: timed out`. 

What instance type do your notebooks use?

Reply:
This was tested to work with t2.medium and t3.medium. Five threads is pretty low so I think it may have to do with where the data is hosted.

Reply:
Because of later errors that I think are due to memory, we might have to set the minimum requirements to be a t3.xlarge.

Reply:
Agreed. Let's put a disclaimer at the top telling folks to switch their NI to a t3.xlarge.

Notebook works for me when t3.xlarge w/ the default 5GB EBS that is added to notebook instances.

Reply:
note added to the Download Data notebook